src/app/users/index.ts

```diff
@@ -36,3 +38,51 @@ export const updateWalletContactAlias = async ({
 
   return user
 }
+
+export const getUserForLogin = async ({
+  userId,
+  ip,
+}: {
+  userId: string
+  ip?: string
+}): Promise<{ domainUser: User; rawUser: UserType } | ApplicationError> => {
+  const repo = UsersRepository()
+
+  const findResult = await repo.findByIdIncludeRaw(userId as UserId)
+  if (findResult instanceof Error) return findResult
+
+  const { domainUser, raw } = findResult
+  domainUser.lastConnection = new Date()
+
+  // IP tracking logic could be extracted into domain
+  const ipConfig = getIpConfig()
+  if (ipConfig.ipRecordingEnabled) {
+    const lastIP: IPType | undefined = domainUser.lastIPs.find(
+      (ipObject: IPType) => ipObject.ip === ip,
+    )
+    if (lastIP) {
+      lastIP.lastConnection = domainUser.lastConnection
+    } else if (ip && ipConfig.proxyCheckingEnabled) {
+      const ipFetcher = IpFetcher()
+      const ipInfo = await ipFetcher.fetchIPInfo(ip)
+      if (!(ipInfo instanceof Error)) {
+        domainUser.lastIPs.push({
+          ip,
+          ...ipInfo,
+          Type: ipInfo.type,
+          firstConnection: domainUser.lastConnection,
+          lastConnection: domainUser.lastConnection,
+        })
+      }
+    }
+  }
+
+  const updateResult = repo.update(domainUser)
+
+  if (updateResult instanceof RepositoryError) {
+    // Don't understand why this cast is necesairy
+    return updateResult as RepositoryError
```

Comment:
This got correctly linted because we didn't `await` on Ln 80
```suggestion
  if (updateResult instanceof RepositoryError) return updateResult
```